modules/network/src/main/java/org/apache/ignite/network/scalecube/ScaleCubeClusterServiceFactory.java

```diff
@@ -83,16 +83,26 @@ public ClusterService createClusterService(
             MessageSerializationRegistry serializationRegistry,
             StaleIds staleIds
     ) {
-        var messageFactory = new NetworkMessagesFactory();
-
         var topologyService = new ScaleCubeTopologyService();
 
+        // Adding this handler as the first handler to make sure that StaleIds is at least up-to-date as any
+        // other component that watches topology events.
+        topologyService.addEventHandler(new TopologyEventHandler() {
+            @Override
+            public void onDisappeared(ClusterNode member) {
+                staleIds.markAsStale(member.id());
```

Comment:
Why don't we call `connectionManager.closeConnectionsWith(member.id());` here as well?

Reply:
First handler is added in the very beginning of `createClusterService` (before start), to make sure that it gets notified first. On the other hand, to close connections, we need an instance of `ConnecitonManager` which is created on start (as it needs configuration). So it doesn't seem we should join them in the same handler.

I agree that it's a bit weird to add this handler in `DefaultMessagingService`, so I moved it to `ScaleCubeClusterServiceFactory` as well.

WDYT?

Reply:
Not ideal, but definitely better than before.